### PR TITLE
Add a couple new features

### PR DESCRIPTION
**Pull request by Fireboyd78**
* Added `-dump_missions` command-line argument
  - Unpacks all mission scripts that weren't already unpacked

- Added vehicle rotation to debug position display

- Added support for custom additional spawn points in Take a Ride (Single Player only)

- Added `REPLAY_EXTRA_DATA` struct to store additional information about the replay (where to spawn, what time of day, weather, etc)

- Fixed 'Player 2' flickering text bug

- Custom spawn point name drawn on Car Select screen

## To-do

* Add & Review additional spawn points for the rest of the cities
  - Recommend 'Safehouse' as last entry for every city
  - Increase number of spawn points? (Current limit is 4)

- Ensure all spawn points can fit big vehicles

- Give appropriate names to all spawn points

- Finalize how to select spawn point from main menu

### Custom spawn point
image

image

### Default spawn point
image

image

### CODE REVIEW

File: src_rebuild/GAME/C/MAIN.C [hunk 1]
```diff
@@ -2637,6 +2637,63 @@ int redriver2_main(int argc, char** argv)
...
 		}
 #endif
+		else if (!_stricmp(argv[i], "-dump_missions"))
+		{
+			char input[32] = { NULL };
+
+			// load from BLK
+			sprintf(input, "MISSIONS\\MISSIONS.BLK");
+
+			if (FileExists(input))
```
Comment (SoapyMan) [inline src_rebuild/GAME/C/MAIN.C:2828]: it needs better to be in separate function... or we need it in OpenDriver2Tools

File: src_rebuild/GAME/FRONTEND/FEMAIN.C [hunk 2]
```diff
@@ -2690,7 +2723,24 @@ int CarSelectScreen(int bSetup)
...
+#define LAST_START_POINT NUM_START_POINTS-1
+	else if (NumPlayers == 1)
+	{
+		if (fePad & 0x4)
+		{
+			FESound(2);
+			if (--wantedStartPos < -1)
+				wantedStartPos = LAST_START_POINT;
+		}
+		else if (fePad & 0x8)
```
Comment (SoapyMan) [inline src_rebuild/GAME/FRONTEND/FEMAIN.C:2736]: maybe just use left/right because L/R shoulder buttons are quite unavailable to keyboard scheme?

Comment (Fireboyd78) [reply src_rebuild/GAME/FRONTEND/FEMAIN.C:2736]: What do you mean? Is it not left shift/right shift?

File: src_rebuild/GAME/C/MAIN.C (hunk 1 above)
Comment (Fireboyd78) [reply src_rebuild/GAME/C/MAIN.C:2828]: Not much code is required to do this. I would only move it to OpenDriver2Tools if you planned on making a (de)compiler tool.

**Comment by Kroc** [comment]
Perhaps also, an option for a "random" spawn point; that is, randomly chosen from the pre-defined spawn-points.

**Pull request closed without merging**